### PR TITLE
Add io.naox.InnerBreeze

### PR DESCRIPTION
**Pull request by waotzi**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
I don't understand the error

**Comment by hfiguiere** [comment]
it is stuck because git ask a password:

```
Username for 'https://github.com': 
```

**Comment by hfiguiere** [comment]
the repository is private.

**Comment by hfiguiere** [comment]
note that even past that I doubt it will build.

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
hi, i'm stuck.

Is this really the issue that's failing the build right now?
```
Error loading desktop file: Failed to find icon: Failed to find icon io.naox.InnerBreeze.png
```

also i don't understand why the image is not found in the desktop file.

I took inspiration from the Myfluttercrypto repo to try and get this working.
https://github.com/flathub/com.github.tglima.Myfluttercrypto/

Thank you

**Comment by hfiguiere** [comment]
is the icon installed? Doesn't look like it.

File: extra/io.naox.InnerBreeze.metainfo.xml
Comment (hfiguiere) [inline extra/io.naox.InnerBreeze.metainfo.xml:1]: this should ship with the upstream distribution

File: extra/io.naox.InnerBreeze.desktop
Comment (hfiguiere) [inline extra/io.naox.InnerBreeze.desktop:1]: this should ship with the upstream distribution

File: extra/io.naox.InnerBreeze.png
Comment (hfiguiere) [inline extra/io.naox.InnerBreeze.png:1]: this should ship with the upstream distribution

File: io.naox.InnerBreeze.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
+  "app-id": "io.naox.InnerBreeze",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "21.08",
```
Comment (hfiguiere) [inline io.naox.InnerBreeze.json:4]: this runtime will be EOL in a few weeks. It's 2 years old.

File: io.naox.InnerBreeze.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          ]
+      },
+      {
+          "name": "InnerBreeze-extra",
+          "buildsystem": "simple",
+          "build-commands": [
+              "install -Dpm644 io.naox.InnerBreeze.desktop /app/share/applications/io.naox.InnerBreeze.desktop",
+              "install -Dpm644 io.naox.InnerBreeze.metainfo.xml /app/share/metainfo/io.naox.InnerBreeze.metainfo.xml"
+          ],
+          "sources": [
```
Comment (hfiguiere) [inline io.naox.InnerBreeze.json:40]:
```suggestion
          "build-commands": [
              "chmod +x inner_breeze",
              "install -D inner_breeze /app/bin/inner_breeze",
              "cp -r data /app/data",
              "cp -r lib /app/lib",
              "install -Dpm644 io.naox.InnerBreeze.desktop /app/share/applications/io.naox.InnerBreeze.desktop",
              "install -Dpm644 io.naox.InnerBreeze.metainfo.xml /app/share/metainfo/io.naox.InnerBreeze.metainfo.xml"
          ],
          "sources": [
              {
                  "type": "archive",
                  "url": "https://github.com/naoxio/inbe/releases/download/1.0.0/linux_x64.tar.gz",
                  "sha256": "0ec1f217e4c27b9691b1089b3578304f78bcfb54689a3002e582a2e4a6f9cbaa"
              },
```

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
i've put 'extra' into the upstream tar.gz and update the file.

looks like there is a new error
```Error loading desktop file: Failed to find icon: Failed to find icon icon```
what icon is it asking for now? i already setup the desktop file :/

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by hfiguiere** [comment]
> i've put 'extra' into the upstream tar.gz and update the file.
> 
> looks like there is a new error `Error loading desktop file: Failed to find icon: Failed to find icon icon` what icon is it asking for now? i already setup the desktop file :/

you desktop file says

`Icon=icon.png`

It's double wrong.
1. icon name. so no extension
2. it's not `icon`

**Comment by waotzi** [comment]
thats the old version, i already updated the tar gz with the correct icon name

**Comment by waotzi** [comment]
the latest hash and targ gz says

Icon=io.naox.InnerBreeze.png

**Comment by hfiguiere** [comment]
that's not what was found in the tarball I donwload and extracted

**Comment by waotzi** [comment]
ohhh wait i messed up. its using the tag not the release

**Comment by waotzi** [comment]
sorry

**Comment by waotzi** [comment]
the sources url is
https://github.com/naoxio/inbe/releases/download/1.0.0/linux_x64.tar.gz

when i use wget it has the correct dekstop icon file.

**Comment by waotzi** [comment]
> that's not what was found in the tarball I downloaded and extracted

why is it downloading a wrong version?

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
hey, I don't understand why It cannot find the binary file inner_breeze

When i try to run the flatpak i get

```
bwrap: execvp inner_breeze: No such file or directory
```

**Comment by waotzi** [comment]
in the build commands i am doing
```
"chmod +x inner_breeze",
"install -D inner_breeze /app/bin/inner_breeze",
```

is that somehow incorrect?

File: io.naox.InnerBreeze.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          "name": "InnerBreeze",
+          "buildsystem": "simple",
+          "build-commands": [
+              "chmod +x inner_breeze",
+              "install -D inner_breeze /app/bin/inner_breeze",
+              "cp -r data /app/data",
+              "cp -r lib /app/lib",
+              "install -Dpm644 extra/io.naox.InnerBreeze.desktop /app/share/applications/io.naox.InnerBreeze.desktop",
+              "install -Dpm644 extra/io.naox.InnerBreeze.metainfo.xml /app/share/metainfo/io.naox.InnerBreeze.metainfo.xml",
+              "install -Dpm644 extra/InnerBreeze.png /app/share/icons/hicolor/256x256/apps/InnerBreeze.png"
```
Comment (hfiguiere) [inline io.naox.InnerBreeze.json:26]: this is the wrong icon name. I'm actually surprise the build doesn't complain.

File: io.naox.InnerBreeze.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{
+  "app-id": "io.naox.InnerBreeze",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "23.08",
+  "sdk": "org.freedesktop.Sdk",
+  "copy-icon": "true",
```
Comment (hfiguiere) [inline io.naox.InnerBreeze.json:6]: this alone does nothing.

File: io.naox.InnerBreeze.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      "--device=dri"
+  ],
+  "command": "inner_breeze",
+  "modules": [
+      {
+          "name": "InnerBreeze",
+          "buildsystem": "simple",
+          "build-commands": [
+              "chmod +x inner_breeze",
+              "install -D inner_breeze /app/bin/inner_breeze",
```
Comment (hfiguiere) [inline io.naox.InnerBreeze.json:21]:
```suggestion
              "install -Dm755 inner_breeze /app/bin/inner_breeze",
```

**Comment by hfiguiere** [comment]
> in the build commands i am doing
> 
> ```
> "chmod +x inner_breeze",
> "install -D inner_breeze /app/bin/inner_breeze",
> ```
> 
> is that somehow incorrect?

the build is incorrect:

```
./inner_breeze: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /nix/store/46m4xx889wlhsdj72j38fnlyyvvvvbyb-glibc-2.37-8/lib/ld-linux-x86-64.so.2, for GNU/Linux 3.10.0, not stripped
```

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

File: io.naox.InnerBreeze.json (hunk 3 above)
Comment (waotzi) [reply io.naox.InnerBreeze.json:25]: hi thanks for the feedback.

why is it wrong? that's what it's called now in the archive, does it have to be named in a specific way?

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
i give up

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

**Comment by waotzi** [comment]
bot, build io.naox.InnerBreeze

File: io.naox.InnerBreeze.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -Dpm644 extra/io.naox.InnerBreeze.desktop /app/share/applications/io.naox.InnerBreeze.desktop",
+                "install -Dpm644 extra/io.naox.InnerBreeze.metainfo.xml /app/share/metainfo/io.naox.InnerBreeze.metainfo.xml",
+                "install -Dpm644 extra/InnerBreeze.png /app/share/icons/hicolor/256x256/apps/InnerBreeze.png",
+                "mv data lib inner_breeze /app/",
+                "mkdir -p /app/bin",
+                "ln -s /app/inner_breeze /app/bin/inner_breeze"
+            ],
+            "sources": [
```
[the remark this answers is not in the document]
Comment (waotzi) [reply io.naox.InnerBreeze.json:26]: thank you for the hint, i removed the nix build and used a more standard github action to build the binary. 

I tested the new flatpak and it seems to be working now

**Comment by waotzi** [comment]
:+1:

**Pull request closed without merging**